### PR TITLE
Use new API for pipeline Q&A astrometry plots

### PR DESCRIPTION
**Pull request by stsci-hack**
The pipeline astrometry Q&A plots were moved out of `quality_analysis.py` into its own (new) module `pipeline_graphics.py`.  All the plots now use the new plotting API from `graph_utils` as well.

In addition, the residuals plots were reset to use the same Y range for all plots, with a minimum range of -0.5 to 0.5.  The plots were also re-ordered on the page to match the order of plots generated by `tweakreg`.  Finally, the vector plot was re-centered so that all the points are visible (from 0 - range) by default.  These modifications to the plot characteristics were all implemented on request by INS.

### CODE REVIEW

File: drizzlepac/haputils/pipeline_graphics.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+                   'fit_results.skew']
+
+RESIDS_COLUMNS = ['residuals.x',
+                  'residuals.y',
+                  'residuals.ref_x',
+                  'residuals.ref_y']
+TOOLSEP_START = '{'
+TOOLSEP_END = '}'
+    
+FIGURE_TOOLS = graph_utils.FIGURE_TOOLS_BASE
```
Comment (mdlpstsci) [inline drizzlepac/haputils/pipeline_graphics.py:119]: For the new interface, this should be removed from this file.  We decided to have a consistent set of tools and not allow the user to add more tools on-the-fly.  This entry should be ignored by the constructor, and I see I added a comment.  However, I did not eliminate the code!  I will fix.  In any case, this line of code should be eliminated.

File: drizzlepac/haputils/pipeline_graphics.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+    num_of_datasets = len(fitCDS.data['index'])
+    print('Number of datasets: {}'.format(num_of_datasets))
+        
+    plot_list = []
+
+    # Data point figures
+    p1 = HAPFigure(title='RMS Values',
+                   x_label="RMS_X (pixels)",
+                   y_label="RMS_Y (pixels)",
+                   tools=FIGURE_TOOLS)
```
Comment (mdlpstsci) [inline drizzlepac/haputils/pipeline_graphics.py:226]: the "tools=FIGURE_TOOLS" should be removed from the parameter list in all places in this code.  You will get the default set of tools defined in graph_utils.py.

File: drizzlepac/haputils/pipeline_graphics.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+HOVER_COLUMNS = ['gen_info.imgname',
+                 'gen_info.instrument',
+                 'gen_info.detector',
+                 'gen_info.filter',                
+                 'header.DATE-OBS',
+                 'header.RA_TARG',
+                 'header.DEC_TARG',
+                 'header.GYROMODE',
+                 'header.EXPTIME',
+                 'fit_results.aligned_to']
```
Comment (mdlpstsci) [inline drizzlepac/haputils/pipeline_graphics.py:94]: Maybe I need to add more docs.  You get all the "gen_info" and "header.*" columns available in the Pandas DataFrame for free in pandas_utils, as well as the 'inst_det' and 'colormap' columns.  While this list can serve the same purpose and are closely related, this is slightly different than generating the hover tooltips.  I did not protect against dups in the hover tools tips list.

You get a few hover tooltips for free in

File: drizzlepac/haputils/pipeline_graphics.py [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+                   'fit_results.nmatches',
+                   'fit_results.skew']
+
+RESIDS_COLUMNS = ['residuals.x',
+                  'residuals.y',
+                  'residuals.ref_x',
+                  'residuals.ref_y']
+TOOLSEP_START = '{'
+TOOLSEP_END = '}'
+
```
Comment (mdlpstsci) [inline drizzlepac/haputils/pipeline_graphics.py:118]: Lines 112 - 117 should be removed from this code as you should not need them.  Build_tool_tips routine lives in graph_utils.py.

File: drizzlepac/haputils/pipeline_graphics.py (hunk 3 above)
Comment (mdlpstsci) [reply drizzlepac/haputils/pipeline_graphics.py:93]: This does not seem to hurt anything even though there will be duplicate columns in the output dataframe.  I can clean this up in the pandas_utils.py.  This is really just a matter of different style of different folks coding. This is not a blocker.

File: drizzlepac/haputils/pipeline_graphics.py (hunk 1 above)
Comment (stsci-hack) [reply drizzlepac/haputils/pipeline_graphics.py]: Removed in 057ba8a.

File: drizzlepac/haputils/pipeline_graphics.py (hunk 2 above)
Comment (stsci-hack) [reply drizzlepac/haputils/pipeline_graphics.py]: Removed (in all locations) in 057ba8a.

File: drizzlepac/haputils/pipeline_graphics.py (hunk 3 above)
Comment (stsci-hack) [reply drizzlepac/haputils/pipeline_graphics.py:93]: Fixed in 057ba8a.

File: drizzlepac/haputils/pipeline_graphics.py (hunk 4 above)
Comment (stsci-hack) [reply drizzlepac/haputils/pipeline_graphics.py]: Fixed in 057ba8a.

**Comment by stsci-hack** [comment]
In addition to addressing the review comments, I also took this opportunity to move the `get_pandas_data()` function from `pipeline_graphics` to `pandas_utils` so that it could be used as a general function.  This new version effectively merges the version from `svm_quality_graphics` and `pipeline_graphics` (formerly, `quality_analysis`).

File: drizzlepac/haputils/pandas_utils.py [hunk 5]
```diff
@@ -35,6 +35,45 @@
 DETECTOR_LEGEND = {'UVIS': 'magenta', 'IR': 'red', 'WFC': 'blue',
                    'SBC': 'yellow', 'HRC': 'black'}
 
+def get_pandas_data(pandas_filename, data_columns, log_level=logutil.logging.NOTSET):
+    """Load the harvested data, stored in a CSV file, into local arrays.
```
Comment (mdlpstsci) [inline drizzlepac/haputils/pandas_utils.py:39]: Change CSV -> HDF5.
Please change CSV to HDF5 all locations as appropriate.

File: drizzlepac/haputils/pipeline_graphics.py [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+    # TODO: include the date from the input data as part of the html filename
+    # Set the output file immediately as advised by Bokeh.
+    output_file(output)
+
+    num_of_datasets = len(fitCDS.data['index'])
+    print('Number of datasets: {}'.format(num_of_datasets))
+        
+    pipeline_tips = build_tooltips(tooltips_list, hover_columns, list(range(0, len(hover_columns))))
+        
+    plot_list = []
```
Comment (mdlpstsci) [inline drizzlepac/haputils/pipeline_graphics.py:173]: plot_list can be deleted.

File: drizzlepac/haputils/pandas_utils.py (hunk 5 above)
Comment (stsci-hack) [reply drizzlepac/haputils/pandas_utils.py]: Addressed in 09f65ce9

File: drizzlepac/haputils/pipeline_graphics.py (hunk 6 above)
Comment (stsci-hack) [reply drizzlepac/haputils/pipeline_graphics.py]: Addressed in 09f65ce9

**Pull request merged**